File: checkDocuments.js
const Typesense = require('typesense');
const _ = require('lodash');
module.exports = (async () => {
  const client = new Typesense.Client({
    nodes: [
      {
        host: 'localhost',
        port: '8108',
        protocol: 'http',
      },
    ],
    apiKey: 'FixO5HtCLHbZhg92OevOl4Yt6SNwZ17TXWfI2RCmsL0GshIu',
  });
  const collections = await client.collections().retrieve();
  const collectionNames = collections.map(collection => {
    return collection.name;
  });
  console.log(collectionNames);
  collectionNames.map(c => {
    try {
      const returnData = await client
        .collections(c)
        .documents()
        .export();
      console.log(
        `No of documents in collection ${c}: `,
        returnData.match(/}/g).length
      );
    } catch (error) {
      console.log(error);
    }
  })
})();

File: checkDocuments.py
import typesense

client = typesense.Client({
  'nodes': [{
    'host': 'localhost',  # For Typesense Cloud use xxx.a1.typesense.net
    'port': '8108',       # For Typesense Cloud use 443
    'protocol': 'http'    # For Typesense Cloud use https
  }],
  'api_key': 'xyz',
  'connection_timeout_seconds': 2
})

return_data = client.collections['blogs'].documents.export()
print('Number of documents: ',len(return_data.split('}'))-1)
